### PR TITLE
wallet: remove deprecated fee bumping by totalFee

### PR DESCRIPTION
**Pull request by jonatack**
Since 0.19, fee-bumping using `totalFee` was deprecated in #15996 and replaced by `fee_rate` in #16727. This changeset removes it.

### CODE REVIEW

File: test/functional/wallet_bumpfee.py [hunk 1]
```diff
@@ -375,21 +346,6 @@ def test_watchonly_psbt(self, peer_node, rbf_node, dest_address):
     rbf_node.unloadwallet("watcher")
     rbf_node.unloadwallet("signer")
 
-def test_rebumping(self, rbf_node, dest_address):
```
Comment (instagibbs) [inline test/functional/wallet_bumpfee.py]: this seems easily updateable to feerate and worthwhile to check?

File: test/functional/wallet_bumpfee.py [hunk 2]
```diff
@@ -375,21 +346,6 @@ def test_watchonly_psbt(self, peer_node, rbf_node, dest_address):
...
-    self.log.info('Test that re-bumping the original tx fails, but bumping successor works')
-    rbfid = spend_one_input(rbf_node, dest_address)
-    bumped = rbf_node.bumpfee(rbfid, {"totalFee": 2000})
-    assert_raises_rpc_error(-4, "already bumped", rbf_node.bumpfee, rbfid, {"totalFee": 3000})
-    rbf_node.bumpfee(bumped["txid"], {"totalFee": 3000})
-
-
-def test_rebumping_not_replaceable(self, rbf_node, dest_address):
```
Comment (instagibbs) [inline test/functional/wallet_bumpfee.py]: this seems easily updateable to feerate and worthwhile to check?

File: test/functional/wallet_bumpfee.py [hunk 3]
```diff
@@ -253,18 +236,6 @@ def test_small_output_with_feerate_succeeds(self, rbf_node, dest_address):
     rbf_node.generatetoaddress(1, rbf_node.getnewaddress())
     assert_equal(rbf_node.gettransaction(rbfid)["confirmations"], 1)
 
-def test_dust_to_fee(self, rbf_node, dest_address):
```
Comment (instagibbs) [inline test/functional/wallet_bumpfee.py]: might want to think about how to convert this test to feerate. It seems like a worthwhile testing gap currently.

File: test/functional/wallet_bumpfee.py [hunk 4]
```diff
@@ -204,14 +195,6 @@ def test_bumpfee_with_descendant_fails(self, rbf_node, rbf_node_address, dest_ad
     rbf_node.sendrawtransaction(tx["hex"])
     assert_raises_rpc_error(-8, "Transaction has descendants in the wallet", rbf_node.bumpfee, parent_id)
 
-def test_small_output_fails(self, rbf_node, dest_address):
```
Comment (instagibbs) [inline test/functional/wallet_bumpfee.py]: this is a correct deletion, feerate arg succeeds where totalFee fails :+1:

**Comment by jonatack** [comment]
Thanks @instagibbs for the quick feedback -- looking into it.

File: test/functional/wallet_bumpfee.py [hunk 5]
```diff
@@ -125,10 +120,6 @@ def test_feerate_args(self, rbf_node, peer_node, dest_address):
     self.sync_mempools((rbf_node, peer_node))
     assert rbfid in rbf_node.getrawmempool() and rbfid in peer_node.getrawmempool()
 
-    assert_raises_rpc_error(-8, "confTarget can't be set with totalFee or fee_rate. Please provide either a confirmation target in blocks for automatic fee estimation, or an explicit fee rate.", rbf_node.bumpfee, rbfid, {"fee_rate":0.00001, "confTarget":1})
```
Comment (kallewoof) [inline test/functional/wallet_bumpfee.py]: This test is still valid, so shouldn't be removed (the message changed though).

**Review by kallewoof — COMMENTED** [review]
Concept ACK and code review with comments 59c1e0a428651ed4c7ea23ad6cfca0429069e232

Verified that the total fee feature is not used anywhere in the code right now (`createBumpTransaction`'s `total_fee` is set to the constant 0 in the one call to the method, in `walletmodel.cpp`, in master).

Agree with @instagibbs points about converting existing tests rather than deleting them.

File: test/functional/wallet_bumpfee.py (hunk 5 above)
Comment (jonatack) [reply test/functional/wallet_bumpfee.py]: Thanks! -- done.

File: test/functional/wallet_bumpfee.py (hunk 3 above)
Comment (jonatack) [reply test/functional/wallet_bumpfee.py]: thanks! -- done

File: test/functional/wallet_bumpfee.py (hunk 2 above)
Comment (jonatack) [reply test/functional/wallet_bumpfee.py]: thanks -- done

File: test/functional/wallet_bumpfee.py (hunk 1 above)
Comment (jonatack) [reply test/functional/wallet_bumpfee.py]: thanks -- done

**Comment by jonatack** [comment]
Thanks @instagibbs and @kallewoof for reviewing. I've updated the existing tests that used totalFee and tried to convert `test_dust_to_fee` too. Could you have a look? Thanks again.

**Comment by jnewbery** [comment]
Concept ACK

File: test/functional/wallet_bumpfee.py [hunk 6]
```diff
@@ -30,14 +30,20 @@
 WALLET_PASSPHRASE = "test"
 WALLET_PASSPHRASE_TIMEOUT = 3600
 
+# Fee rates
+INSUFFICIENT = 0.00001000
+ECONOMY      = 0.00050000
```
Comment (kallewoof) [inline test/functional/wallet_bumpfee.py]: μNit (ignore unless you rebase or do further updates): `ECONOMICAL`?

**Review by kallewoof — COMMENTED** [review]
Re-ACK 9dcee5fcd8ba9aba9fadfd4de008d846f022dfc6

File: test/functional/wallet_bumpfee.py (hunk 6 above)
Comment (jonatack) [reply test/functional/wallet_bumpfee.py]: Good catch! `git grep -ni econom` shows the codebase and `sendtoaddress` rpc help all use "economical".

**Comment by jonatack** [comment]
Thanks for reviewing @kallewoof. Updated constant name as per `git diff 9dcee5f..abf2ab15`

File: test/functional/wallet_bumpfee.py (hunk 6 above)
Comment (jonatack) [reply test/functional/wallet_bumpfee.py]: Good catch! git grep -ni econom shows the codebase and rpc help use that. Done.

**Comment by kallewoof** [comment]
Re-ACK abf2ab1

<details>
  <summary>git diff 9dcee5f</summary>

```diff
diff --git a/test/functional/wallet_bumpfee.py b/test/functional/wallet_bumpfee.py
index 5aa6a1b6f..38c980775 100755
--- a/test/functional/wallet_bumpfee.py
+++ b/test/functional/wallet_bumpfee.py
@@ -30,9 +30,9 @@ from test_framework.util import (
 WALLET_PASSPHRASE = "test"
 WALLET_PASSPHRASE_TIMEOUT = 3600

-# Fee rates
+# Fee rates (in BTC per 1000 vbytes)
 INSUFFICIENT = 0.00001000
-ECONOMY      = 0.00050000
+ECONOMICAL   = 0.00050000
 NORMAL       = 0.00100000
 HIGH         = 0.00500000
 TOO_HIGH     = 1.00000000
@@ -379,7 +379,7 @@ def test_watchonly_psbt(self, peer_node, rbf_node, dest_address):
 def test_rebumping(self, rbf_node, dest_address):
     self.log.info('Test that re-bumping the original tx fails, but bumping successor works')
     rbfid = spend_one_input(rbf_node, dest_address)
-    bumped = rbf_node.bumpfee(rbfid, {"fee_rate": ECONOMY})
+    bumped = rbf_node.bumpfee(rbfid, {"fee_rate": ECONOMICAL})
     assert_raises_rpc_error(-4, "already bumped", rbf_node.bumpfee, rbfid, {"fee_rate": NORMAL})
     rbf_node.bumpfee(bumped["txid"], {"fee_rate": NORMAL})

@@ -387,7 +387,7 @@ def test_rebumping(self, rbf_node, dest_address):
 def test_rebumping_not_replaceable(self, rbf_node, dest_address):
     self.log.info('Test that re-bumping non-replaceable fails')
     rbfid = spend_one_input(rbf_node, dest_address)
-    bumped = rbf_node.bumpfee(rbfid, {"fee_rate": ECONOMY, "replaceable": False})
+    bumped = rbf_node.bumpfee(rbfid, {"fee_rate": ECONOMICAL, "replaceable": False})
     assert_raises_rpc_error(-4, "Transaction is not BIP 125 replaceable", rbf_node.bumpfee, bumped["txid"],
                             {"fee_rate": NORMAL})

@@ -464,7 +464,7 @@ def test_change_script_match(self, rbf_node, dest_address):
     assert_equal(len(change_addresses), 1)

     # Now find that address in each subsequent tx, and no other change
-    bumped_total_tx = rbf_node.bumpfee(rbfid, {"fee_rate": ECONOMY})
+    bumped_total_tx = rbf_node.bumpfee(rbfid, {"fee_rate": ECONOMICAL})
     assert_equal(change_addresses, get_change_address(bumped_total_tx['txid']))
     bumped_rate_tx = rbf_node.bumpfee(bumped_total_tx["txid"])
     assert_equal(change_addresses, get_change_address(bumped_rate_tx['txid']))
```
</details>

File: src/wallet/rpcwallet.cpp [hunk 7]
```diff
@@ -3342,24 +3342,19 @@ static UniValue bumpfee(const JSONRPCRequest& request)
...
                 "An opt-in RBF transaction with the given txid must be in the wallet.\n"
                 "The command will pay the additional fee by reducing change outputs or adding inputs when necessary. It may add a new change output if one does not already exist.\n"
-                "If `totalFee` (DEPRECATED) is given, adding inputs is not supported, so there must be a single change output that is big enough or it will fail.\n"
                 "All inputs in the original transaction will be included in the replacement transaction.\n"
                 "The command will fail if the wallet or mempool contains a transaction that spends one of T's outputs.\n"
                 "By default, the new fee will be calculated automatically using estimatesmartfee.\n"
                 "The user can specify a confirmation target for estimatesmartfee.\n"
-                "Alternatively, the user can specify totalFee (DEPRECATED), or fee_rate (" + CURRENCY_UNIT + " per kB) for the new transaction .\n"
+                "Alternatively, the user can specify a fee_rate (" + CURRENCY_UNIT + " per kB) for the new transaction .\n"
```
Comment (laanwj) [inline src/wallet/rpcwallet.cpp]: nit: If you're changing this line anyway, please remove the space before the `.`.

Comment (jonatack) [reply src/wallet/rpcwallet.cpp]: thanks -- good eye!

**Comment by jonatack** [comment]
Fixup to remove an extra space as per `git diff abf2ab1..c3857c5`.

**Comment by laanwj** [comment]
ACK c3857c5fcb21836ddc1b79a6b19cffe562cade10

Only difference before is the space doc change:
```patch
 $ git diff abf2ab1 c3857c5
diff --git a/src/wallet/rpcwallet.cpp b/src/wallet/rpcwallet.cpp
index 7df846e0fc57053d9c082ff232804d7a83e29bbd..05342472bbeeb6484b947dff649810a55723f563 100644
--- a/src/wallet/rpcwallet.cpp
+++ b/src/wallet/rpcwallet.cpp
@@ -3346,7 +3346,7 @@ static UniValue bumpfee(const JSONRPCRequest& request)
                 "The command will fail if the wallet or mempool contains a transaction that spends one of T's outputs.\n"
                 "By default, the new fee will be calculated automatically using estimatesmartfee.\n"
                 "The user can specify a confirmation target for estimatesmartfee.\n"
-                "Alternatively, the user can specify a fee_rate (" + CURRENCY_UNIT + " per kB) for the new transaction .\n"
+                "Alternatively, the user can specify a fee_rate (" + CURRENCY_UNIT + " per kB) for the new transaction.\n"
                 "At a minimum, the new fee rate must be high enough to pay an additional new relay fee (incrementalfee\n"
                 "returned by getnetworkinfo) to enter the node's mempool.\n",
                 {
```

**Comment by jonatack** [comment]
Thanks -- will add a release note on the dev wiki.

**Comment by jnewbery** [comment]
utACK c3857c5fcb21836ddc1b79a6b19cffe562cade10

**Pull request merged**